### PR TITLE
 [SPARK-39029][PYTHON][TESTS] Improve the test coverage for pyspark/broadcast.py

### PR DESCRIPTION
**Pull request by pralabhkumar**
### What changes were proposed in this pull request?
This PR add test cases for broadcast.py

### Why are the changes needed?
To cover corner test cases and increase coverage

### Does this PR introduce _any_ user-facing change?
No - test only

### How was this patch tested?
CI in this PR should test it out

### CODE REVIEW

File: python/pyspark/tests/test_broadcast.py [hunk 1]
```diff
@@ -99,6 +101,30 @@ def test_broadcast_value_against_gc(self):
         finally:
             b.destroy()
 
+    def test_broadcast_when_sc_none(self):
+        # SPARK-39029 : Test case to improve test coverage of broadcast.py
+        # It test the case when sc is none and Broadcast is called at executor
```
Comment (HyukjinKwon) [inline python/pyspark/tests/test_broadcast.py:106]:
```suggestion
        # It tests the case when SparkContext is none and Broadcast is called at executor
```

File: python/pyspark/tests/test_broadcast.py [hunk 2]
```diff
@@ -99,6 +101,30 @@ def test_broadcast_value_against_gc(self):
...
+        conf = SparkConf()
+        conf.setMaster("local-cluster[2,1,1024]")
+        self.sc = SparkContext(conf=conf)
+        bs = self.sc.broadcast([10])
+        bs_sc_none = Broadcast(sc=None, path=bs._path)
+        self.assertEqual(bs_sc_none.value, [10])
+
+    def test_broadcast_for_error_condition(self):
+        # SPARK-39029: Test case to improve test coverage of broadcast.py
+        # It test the case when broadcast should raise error .
```
Comment (HyukjinKwon) [inline python/pyspark/tests/test_broadcast.py:116]:
```suggestion
        # It tests the case when broadcast should raise error .
```

File: python/pyspark/tests/test_broadcast.py [hunk 3]
```diff
@@ -99,6 +101,30 @@ def test_broadcast_value_against_gc(self):
...
+        conf = SparkConf()
+        conf.setMaster("local-cluster[2,1,1024]")
+        self.sc = SparkContext(conf=conf)
+        bs = self.sc.broadcast([1])
+        with self.assertRaisesRegex(pickle.PickleError, "Could.*not.*serialize.*broadcast"):
+            self.sc.broadcast(self.sc)
+        with self.assertRaisesRegex(Exception, "RuntimeError.*Broadcast.*destroyed.*driver"):
```
Comment (HyukjinKwon) [inline python/pyspark/tests/test_broadcast.py:123]: Can we catch narrower exception?

**Review by HyukjinKwon — COMMENTED** [review]
Looks pretty good otherwise. Thanks for working on this.

**Comment by pralabhkumar** [comment]
@HyukjinKwon  Please review the changes . Build is failing may be because of unrelated changes 

Execution halted
                    ------------------------------------------------
      Jekyll 4.2.1   Please append `--trace` to the `build` command 
                     for any additional information or backtrace. 
                    ------------------------------------------------
/__w/spark/spark/docs/_plugins/copy_api_dirs.rb:147:in `<top (required)>': R doc generation failed (RuntimeError)

**Comment by HyukjinKwon** [comment]
Yeah, I don't think that's related.

Merged to master.

**Pull request closed without merging**